CONTRIBUTING.md

```diff
@@ -31,15 +33,31 @@ It is intended to be migrated to Jekyll markdown and to be published with github
 
 ## Contributing code
 
+Before development is started, a [JIRA task or bug](#working-with-jira) should have been created for the issue.
 All development is managed using [pull requests](#pull-requests-workflow).
-[Commits](#commits) must be in logical, consistent units and have a good commit message.
+[Commits](#commits) must be in logical, consistent units and have a good commit message that includes the JIRA key of the corresponding issue.
```

Comment:
IMO commit messages should only contain the JIRA key if they actually close the issue. So in a PR, you'd have only one commit (usually the last one) with a JIRA key.

If that is the case, then it's a bit too subtle to explain in this short sentence, so I'd leave it unchanged.

Reply:
We should also state how to mention Jira key (e.g: in commit title, as last lin of commit).
```suggestion
[Commits](#commits) must be in logical, consistent units and have a good commit message that includes the JIRA key of the corresponding issue as last line.
```

Reply:
This section of the CONTRIBUTING.md file is just a general summary. Details are in the next chapter ("Working with JIRA"), so I'll just put any clarification there.
I also generally prefer every commit to have the JIRA key. It makes the linkage easier and more explicit, and makes it easier to search for commits as Ran mentioned.
As for where to put them, I honestly don't care much, other than that it probably should not go into the first line because we already have the category in there so it would start to become a bit much